### PR TITLE
Add ic and jct information to step intersection

### PR DESCRIPTION
**Pull request by abhishek1508**
Add `IC` and `JCT` information to `StepIntersection`.

### CODE REVIEW

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java
Comment (dzinad) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java:55]:
```suggestion
   * @param json a formatted valid JSON string defining an InterChange
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Junction.java
Comment (dzinad) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Junction.java:55]:
```suggestion
   * @param json a formatted valid JSON string defining an Junction
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Junction.java
Comment (dzinad) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Junction.java:71]:
```suggestion
     * The name of the junction. Optionally included if data is available.
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Junction.java
Comment (dzinad) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Junction.java:73]:
```suggestion
     * @param name junction name
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java
Comment (dzinad) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:510]: It returns a builder, not "an object containing information about routing and passing IC(s)..."

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java
Comment (dzinad) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:521]: It returns a builder, not "an object containing information about routing and passing JCT(s)..."

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java
Comment (dzinad) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java:9]: Maybe we should also check the default object? That "name" is null?

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/JunctionTest.java
Comment (dzinad) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/JunctionTest.java:9]: Maybe we should also check the default object? That "name" is null?

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/StepIntersectionTest.java
Comment (dzinad) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/StepIntersectionTest.java:166]: How about a case when junction is null?

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/StepIntersectionTest.java
Comment (dzinad) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/StepIntersectionTest.java:179]: How about a case when interchange is null?

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java
Comment (abhishek1508) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java:9]: Do you mean that `InterChange` object is not null even though `interchange.name` is null?

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/JunctionTest.java
Comment (abhishek1508) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/JunctionTest.java:9]: Do you mean that `Junction` object is not null even though `junction.name` is null?

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java
Comment (dzinad) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java:9]: Yes. You say in the docs that name is optional.

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/JunctionTest.java
Comment (dzinad) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/JunctionTest.java:9]: Yes. You say in the docs that name is optional.

Comment (abhishek1508) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/JunctionTest.java:9]: Done

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java
Comment (abhishek1508) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java:9]: Done.

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java
Comment (dzinad) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/InterChangeTest.java:23]: I meant something different. :)
```
InterChange interChange = InterChange.builder().build();
assertNotNull(interChange.name());
```

**Review by dzinad — APPROVED** [review]

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java:15]:
```suggestion
public abstract class Interchange extends DirectionsJsonObject {
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java:11]:
```suggestion
/*
```
Same everywhere else.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:253]:
```suggestion
  public abstract InterChange interchange();
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java
Comment (abhishek1508) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java:11]: Is this supposed to be hidden? Otherwise this should be a Javadoc comment for class

Comment (abhishek1508) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/InterChange.java]: Done. Made the passive maneuvers hidden.

**Comment by abhishek1508** [comment]
@LukasPaczos I need an approval before I can merge

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:254]:
```suggestion
  public abstract Interchange interchange();
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:517]:
```suggestion
    public abstract Builder interchange(@Nullable Interchange interchange);
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:508]:
```suggestion
     * An object containing information about routing and passing interchanges along the route.
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:257]:
```suggestion
   * An object containing information about routing and passing junctions along the route.
```

**Review by LukasPaczos — COMMENTED** [review]
Some name changes still needed (please update all occurrences, I haven't highlighted all of them), otherwise, LGTM.

**Pull request merged**
